src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1927,11 +2002,16 @@ GenTree* Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cm
 
                         default:
                         {
-                            unreached();
+                            maskIntrinsicId = NI_AVX512F_NotMask;
```

Comment:
There's already a couple lengthy comment above this code elaborating on the overall needs and the needs of this path in particular.

For the cases where we have a partial mask, we need to invert the comparison. This is because `matches` become `1` and bits are otherwise `0`, including any bits that are unused in the comparison.

Consider for example if we had `Vector128<int>` and so we have `4-bits` in the mask, but the mask instruction always checks at least `8-bits`. If we did `CompareEqual` and all elements matched, we'd still get `0b0000_1111` and so we couldn't easily check "are all matching". However, if we invert the comparison and do `CompareNotEqual` we get `0b0000_0000` for all matches and can then trivially check "are all matching".

This is meant to handle the general case of "other mask intrinsic" where we don't have a well known inverse. Notably I think I forgot a bit of code here, we need to either invert just the lower `n` bits or always set the upper 4-bits. Will fix to ensure just that is happening